packages/firestore/src/local/memory_remote_document_cache.ts

```diff
@@ -183,49 +194,63 @@ export class MemoryRemoteDocumentCache implements RemoteDocumentCache {
   }): RemoteDocumentChangeBuffer {
     // `trackRemovals` is ignores since the MemoryRemoteDocumentCache keeps
     // a separate changelog and does not need special handling for removals.
-    return new MemoryRemoteDocumentCache.RemoteDocumentChangeBuffer(this);
+    return new MemoryRemoteDocumentChangeBuffer(this);
   }
 
   getSize(txn: PersistenceTransaction): PersistencePromise<number> {
     return PersistencePromise.resolve(this.size);
   }
+}
 
-  /**
-   * Handles the details of adding and updating documents in the MemoryRemoteDocumentCache.
-   */
-  private static RemoteDocumentChangeBuffer = class extends RemoteDocumentChangeBuffer {
-    constructor(private readonly documentCache: MemoryRemoteDocumentCache) {
-      super();
-    }
+/**
+ * Creates a new memory-only RemoteDocumentCache.
+ *
+ * @param indexManager A class that manages collection group indices.
+ * @param sizer Used to assess the size of a document. For eager GC, this is expected to just
+ * return 0 to avoid unnecessarily doing the work of calculating the size.
+ */
+export function newMemoryRemoteDocumentCache(
+  indexManager: IndexManager,
+  sizer: DocumentSizer
+): MemoryRemoteDocumentCache {
+  return new MemoryRemoteDocumentCacheImpl(indexManager, sizer);
```

Comment:
TypeScript noobie question: How does this compile since `MemoryRemoteDocumentCacheImpl` does not implement `MemoryRemoteDocumentCache`?

Reply:
TypeScript interfaces are enforced by duck-typing - if a type happens to implement all members of an interface, it is considered identical to a type that explicitly implements the interface. Explicitly implementing comes in handy when the methods are renamed or added as it will produce much saner errors.